cmake/sanitizers.cmake

```diff
@@ -5,8 +5,12 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" ST
     option(ADDRESS_SANITIZER "Enable the address sanitizer" OFF)
     message(STATUS "  + ADDRESS_SANITIZER                     ${ADDRESS_SANITIZER}")
     if(ADDRESS_SANITIZER)
-        add_compile_options(-fsanitize=address -fno-omit-frame-pointer)
-        link_libraries(-fsanitize=address -fno-omit-frame-pointer)
+        add_compile_options(
+           -fsanitize=address
+           -fsanitize-address-use-after-scope
```

Comment:
This flag is meant to be enabled by default per https://github.com/google/sanitizers/wiki/AddressSanitizerUseAfterScope#introduction. Was it missing somewhere?

Reply:
Ah, you are right. I was following the tips here:

https://github.com/google/sanitizers/wiki/AddressSanitizer


> Q: Can I run AddressSanitizer with more aggressive diagnostics enabled?

>A: Yes! In particular you may want to enable

 >CFLAGS += -fsanitize-address-use-after-scope
 >ASAN_OPTIONS=strict_string_checks=1:detect_stack_use_after_return=1:check_initialization_order=

But it seems like that information is outdated. Will update.

Reply:
I completely missed that part. 😁 Hopefully they default to the stricter check now, but who knows. I checked an old Travis build, but I guess we don't print anything about the flags there.